nixos/modules/system/boot/stage-1.nix

```diff
@@ -30,20 +30,34 @@ let
   # mounting `/`, like `/` on a loopback).
   fileSystems = filter utils.fsNeededForBoot config.system.build.fileSystems;
 
-  # A utility for enumerating the shared-library dependencies of a program
-  findLibs = pkgs.buildPackages.callPackage ./find-libs.nix {};
-
   extraUnitsObjects = map (u: { object = u; }) (builtins.attrValues config.boot.initrd.extraUnits
     ++ lib.concatLists (lib.mapAttrsToList (_: builtins.attrValues) config.boot.initrd.unitOverrides));
   systemdUnits = pkgs.callPackage ./systemd-units.nix {
     inherit systemd;
     inherit (config.boot.initrd) extraUnits unitOverrides;
+    systemdPackages = config.boot.initrd.systemd.packages;
   };
 
   fstab = pkgs.writeText "fstab" (lib.concatMapStringsSep "\n"
     ({ fsType, mountPoint, device, options, ... }:
       "${device} /sysroot${mountPoint} ${fsType} ${lib.concatStringsSep "," options}") fileSystems);
 
+  groups = [
```

Comment:
What is the source of this list? When do we add groups here? Is that the list of groups that systemd somewhere specifies?

Reply:
udev warned about non-existent groups in the journal, so I threw these in, in hopes they'd fix whatever I was debugging at the time. I don't think they're necessary at all, and can probably be removed.